### ISSUE TITLE
Transfer performance remains significantly degraded in 4.0.1 compared to 3.0.0

### ISSUE THREAD

**Issue by ilikepeaches** [transmission/transmission#5003]
### What is the issue?

Opening a new issue to continue the discussion from https://github.com/transmission/transmission/issues/4849

Per the instructions in the old issue, I tried 4.0.1 to see if there were noticeable transfer performance improvements.  Answer: definitely not.

I ambled back over to 3.0.0 to see how big the difference was and watched my upload & download skyrocket to a level I've not seen since switching to 4, then stay at that level.

A well-seeded torrent on 4 is like: transfer at full speed for a few seconds, then sit at 0K/sec for five to ten seconds, then a quick burst of activity again, then another lengthy period of nothing, then...

The "Active" view (which works properly in 4, yay, and never did in 3) shows that the bursts happen at the same time for all active torrents, and that the periods of zero activity are similarly shared.

On 3, it's: transfer consistently, with no periods of zero up/down.

### Which application of Transmission?

macOS app

### Which version of Transmission?

4.0.1

**Comment by ckerr** [comment]
Same number of peers between the two versions?

If some peers are missing in 4, is there any pattern in which ones are missing or which ones are still there, e.g. connected via TCP or uTP?

**Comment by ilikepeaches** [comment]
Apparently fewer peers in 4.
Per conversation in the previous thread, I turned off μTP.
Exact same torrent, moments apart.
Exact same save location, etc.  No I/O bottlenecks here.
Mac GUI client 4.0.1 vs. 3.0.0

Periods of no activity on 4:
![Screenshot 2023-02-22 at 9 42 20 PM](https://user-images.githubusercontent.com/44431584/220811517-a8576329-8349-49fa-a279-2c1c31aaa2ea.png)

Periods of activity on 4:
![Screenshot 2023-02-22 at 9 47 12 PM](https://user-images.githubusercontent.com/44431584/220811528-c4fed88f-f416-4fe0-91f0-b507c128688c.png)

Consistent behavior on 3:
![Screenshot 2023-02-22 at 9 38 13 PM](https://user-images.githubusercontent.com/44431584/220811589-76247293-eb43-4221-becf-0e4ca2b45174.png)

**Comment by Coding4Coffee** [comment]
Can you try with the following torrent:

[Ubuntu 20.04.5 LTS](https://releases.ubuntu.com/20.04/ubuntu-20.04.5-desktop-amd64.iso.torrent)

Also please post the following:

1) Picture of Transmission->Settings->Bandwidth
<img width="752" alt="Screenshot 2023-02-23 at 03 05 40" src="https://user-images.githubusercontent.com/25304836/220812818-2fd98156-da88-48f0-aac4-c82c6539bf78.png">

2) Picture of Transmission->Settings->Peers
<img width="752" alt="Screenshot 2023-02-23 at 03 06 01" src="https://user-images.githubusercontent.com/25304836/220812882-b65381ad-63f5-41bb-9e18-ec31429306d3.png">

3) Picture of Transmission->Settings->Network
<img width="752" alt="Screenshot 2023-02-23 at 03 06 34" src="https://user-images.githubusercontent.com/25304836/220812941-5259f650-26fc-4d0b-b322-3250a6f2e670.png">

3) 'Command+3' debug log whilst on Transmission 4. You may need to scrub your home directory name in the logs.

[T401-Debug.txt](https://github.com/transmission/transmission/files/10809900/T401-Debug.txt)

This is the full debug log for the download of the Ubuntu torrent, my speeds were consistent, at 10MB/s which is utilising the full bandwidth of my connection. Although for some reason UPnP fails although it is enabled on my Orbi router. It did not impact my speeds, however.

**Comment by ilikepeaches** [comment]
FWIW, I exclusively use Transmission with private torrents, so I have little experience with normal performance on public torrents or perspective to offer about PEX, DHT, etc. settings.

I spent some time looking at the debug-level log for the same ~1k private torrents, switching between Transmission 3 and Transmission 4 (still observing significantly higher transfer rates on 3).  What stands out is that the Transmission 4 log is absolutely littered with this error:

```do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error```

which
1. does not appear in the Transmission 3 logs at all
2. I see mentioned alongside performance complaints in https://github.com/transmission/transmission/issues/3111 and https://github.com/transmission/transmission/issues/4046 — where @ckerr said "That one's probably worth opening a separate ticket for. IDK what's causing it and it could be nothing, but best to make sure."

but (unless Google fails me) it seems like no further investigation of those errors has occurred?

**Comment by lllvvv** [comment]
This is a very weird bug!

Here's what I tried:
1) I gathered two large torrents, one public (4GB) and one private (6GB), both of them are well seeded (I was seeing 100 seeds for each);
2) Started the private torrent, and for the first 60 seconds I was seeing the expected speeds (25MB/s, but with some variation). After that, speeds dropped to 6~10MB/s and kept there for a while.
3) Paused the private torrent, and added the public one. All the peers were found rather quick, but I was still seeing the lower speeds. Paused the public torrent.
4) Quit transmission, unload and load the networking module, reconnect;
5) Unpause the private torrent: it finished the remaining 80% of the torrent always fully utilizing bandwidth;
6) Unapuse the public torrent: same story.

I was watching the debug log while this was happening, and while I do see a lot of the "do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error", they were happening regardless of the throughput of the torrent - I mean, someone should still look into this, but it's not related to the performance problems.

I'm using 4.0.1-dev (c43ed9a8a1) of the gtk client. These are my settings:

![2023-02-25-093654](https://user-images.githubusercontent.com/11249333/221350135-d9f00293-21c6-4bbe-a10e-db461d9de233.jpg)

**Comment by kubrickfr** [comment]
I'm seeing the same thing on android with the Termux package.
It must be an ARM problem. 

I see very few peers for torrents that are well seeded, and as a result transfer speeds are super slow.

**Comment by GaryElshaw** [comment]
The discrepancy i'm seeing is in the upload speeds when seeding, it's often in double-digits and lower using the GTK and Qt clients. When downloading, the upload speeds are significanly better.

**Comment by EDavidzar** [comment]
Hello to everybody, with qnap nas arm64, dowloading 1GB optic fiber connection, with router open transmission port  and not upnp not utp not port forwarding in settings, transmission closed port but port is open in router, Transmission 3.0 is very faster 20MB/s - 60MB/s but Transmission 4 4.0.0 and 4.0.1 version are very slow 4MB/s -6MB/s, always with public trackers, seems to be global problem and not only on mac, windows not tested. Greetings

**Comment by ilikepeaches** [comment]
FWIW, I've observed & tested this issue by flipping back and forth between Transmission 3 & 4 on an **Intel** Mac, so I don't think it's an ARM-only issue.

**Comment by EDavidzar** [comment]
I thought that maybe we should create a prototype/system to standardise the transmission tests, with certain parameters and certain download files, I don't know if you are ok with this in order to be able to make standardised tests.

**Comment by EDavidzar** [comment]
Windows 10 testing ...  same parameters, seems to be OK high download rate 20MB/s and upload rate 20Mb/s aprox.

**Comment by sinjinsmythe** [comment]
I have the same issue since upgrading to 4.01 on my Synology 716+ NAS with Intel Celeron processor running inside a docker container.
Downloads typically not going above 250k/sec

**Comment by GaryElshaw** [comment]
I'm seeing a difference between macOS and the Linux clients (GTK and Qt), with macOS having better conectivity during uploads. I'm still trying to think what the commonalities on linux might be, and wondering if something like dbus might be the culprit?

**Comment by tollfree** [comment]
<img width="894" alt="Screenshot 2023-02-27 at 18 49 16" src="https://user-images.githubusercontent.com/5585001/221615564-848e5f34-18f9-4cb3-aa00-b0922b1799e6.png">
There are similar errors for every torrent, existed before or just added, in 4.0.1.

**Comment by ilikepeaches** [comment]
Seriously, it's 9-12MiB/sec download on Transmission 4, with the transfer rates breathing in & out / up & down to zero.  Then I quit and start Transmission 3 instead, where I immediately get 25-60MiB/sec download on the same torrents.

**Comment by GaryElshaw** [comment]
I thought i'd have a look at tuning my ethernet to see if that was where the problem might lie (it's decent and i'm using a Cat 8 cable). I'm currently uploading to two peers at 1 - < 200 K/s.

NIC output looks fine to me.

NIC statistics:
     tx_packets: 879447
     rx_packets: 277644
     tx_errors: 0
     rx_errors: 0
     rx_missed: 2
     align_errors: 0
     tx_single_collisions: 0
     tx_multi_collisions: 0
     unicast: 276983
     broadcast: 374
     multicast: 287
     tx_aborted: 0
     tx_underrun: 0

**Comment by ilikepeaches** [comment]
Unfortunately, I think this issue should be top priority.  It makes Transmission 4 unusable as a daily driver for me.

:-(

**Comment by Coding4Coffee** [comment]
@ilikepeaches Can you please test with the Ubuntu torrent test case and report back. Try it with the following:

a) All torrents paused and only the Ubuntu torrent downloading.
b) All your torrents active with seeding going on in the background, whilst downloading.

We also need responses for: 

1. How many torrents do you have?
2. How many torrents are actively downloading when the issue occurs? 
3. Does it happen when only one torrent is active, and the rest paused or when others are seeding?
4. What operating system are you using?
5. How much RAM do you have?
6. What type of storage is the download writing to? NVME/SSD/HDD/NAS?
7. Are you saving to an externally connected drive or built in?
8. How full is your hard drive?
9. Does the speed drop occur at the beginning, middle, or end of the download? Or is it an issue throughout?
10. What are your settings for Bandwidth, Network and Peers? Attach screenshots.
11. Screenshot of your trackers tab (Are they connecting and announcing okay, or any failures?)
12. Screenshot of your peer tab (How many peers are connecting from the swarm)
13. Attach your debug log `Command + 3` Note: Scrub IPs and home directory name if needed and other PIA you find necessary.

> I thought that maybe we should create a prototype/system to standardise the transmission tests, with certain parameters and certain download files, I don't know if you are ok with this in order to be able to make standardised tests.

I agree with @EDavidzar, this would be handy. It's always going to be hard to have consistent results due to a lot of factors out of control. However, we can use the following publicly available torrents, as public tracker test cases: 

**High availability and web seeded (As of 1st March 2023)**

[Ubuntu 20.04.5 LTS](https://releases.ubuntu.com/20.04/ubuntu-20.04.5-desktop-amd64.iso.torrent)
[Kali 2022.4](https://cdimage.kali.org/kali-2022.4/kali-linux-2022.4-installer-amd64.iso.torrent)

**Medium/Low availability and non-web seeded (As of 1st March 2023)**

[Slackware](http://www.slackware.com/torrents/slackware64-14.2-install-dvd.torrent)

If this issue occurs with slower storage, it may be an issue in the I/O code on certain operating systems.

I'm on a MBP M1, 64GB RAM, with 4TB SSDs (90% full) and have no issues with the above 3 torrents on `4.0.1-dev (e9eb70b50f)` 



Off topic: @GaryElshaw Cat 8? Do you live in a data centre? 😜

**Comment by ilikepeaches** [comment]
@Coding4Coffee, I've already answered the bulk of these questions, and most of the others seem unrelated and off-topic.  I am not the only person observing these issues.  The Ubuntu test case you propose is completely irrelevant to me, as I do not use Transmission for public torrents and don't see any reason to introduce the many additional variables and layers of complexity involved in adding them to the mix.  I decline to publicly share logs, scrubbed or otherwise, especially when the person demanding them is, as far as I can tell, just some random person with no contributions to any repo.  I will wait for actual project contributors to engage on the potential hypotheses that have already been raised here (e.g. I already mentioned overly deferential behavior toward external I/O and have pointed to log messages also identified in previous conversations about performance issues) before I take lengthy assignments from anyone else.

**Comment by ilikepeaches** [comment]
@Coding4Coffee You are not a maintainer or contributor on this project and therefore could step a lot more delicately when taking it upon yourself to demand long laundry lists of questionably probative information ("attach screen shots") from others.  To your sneering & uncalled-for comment about Google, I only say: don't believe everything you read on the Internet.

Back to work: please confirm that you are testing Transmission 4 against Transmission 3 **directly**, as I have been.  "No issues" with three torrents by itself tells us nothing, as this issue focuses on Transmission 4's transfer performance when compared to the significantly higher transfer performance of Transmission 3 **when all other variables are controlled**.  Most of the conversation in this thread and its forebear acknowledges some transfer happening — it's just significantly degraded when compared one-to-one against Transmission 3's performance under the identical combination of operating system/torrents/trackers/connection/settings/drives/system load/etc.  The reason none of your _thirteen_ questions are likely to lead anywhere fruitful is that my testing environment accounts for all of them — by holding them constant and only changing the version of Transmission being run.  I am testing — on the same 1k+ private torrents on a mix of heavily utilized / nearly full HDD & SSD on Mac/Intel — by simply quitting one version of Transmission and launching the other.  Nothing else varies.  The difference in performance between the two versions is clear & consistent.

I bet there's a more programmatic way to approach this mystery — probably having to do with generating stacktraces/sysdiagnoses or other concrete output and sharing that information directly with the appropriate project representatives — and I ask & invite people like @ckerr to guide us away from whatever this has become and toward that sort of productive investigation.

**Comment by doomtop** [comment]
I have this same issue with v4 (including 4.0.1) rendering it unusable for me. I've reverted to v3 to fix the problem.

**Comment by JustinTArthur** [comment]
@ilikepeaches and @doomtop one thing that would be of interest to this is to know whether you're on a dual-stack node (both IPv4 and IPv6). If you, your DNS resolver, and your private tracker are all dual-stack, the behavior between Transmission  has changed between v3 and v4. It's just one thing out of many things worth checking. This one's easier than looking for something blocking the event loop in code—something that's still a potential culprit.

In the dual-stack scenario, v3 would announce you to _either_ the IPv4 or IPv6 IP of the tracker. Which one was up to libcurl and your OS' name-resolution stack. If you were lucky, it would announce to the IPv4, give it the `ipv6=` param with your additional IPv6 address (if transmission could find your external one), and if the tracker was old enough to still accept `ipv6=`, you had a dual-stack announcement.

In v4, you will announce once from an IPv4 source address and once from your IPv6 source address. This is ultimately better and should give you a higher chance of being peer-connectible when a tracker has both IPs/ports for you. HOWEVER, this can make things worse for you when a private tracker has an IP+port-per-user limit on a given torrent, and many do. Some trackers will reject more than 3 IP/port combos per authenticated user. If you restart Transmission with port randomization enabled, you suddenly have 4 different IP/ports you've announced for a given torrent. The tracker might discard the oldest entry, LRU-style or it might reject your newest entry, making you un-discoverable/un-connectible by that route. When this rejection happens, you'll have to wait for that tracker's flush interval for it expire those old entries, e.g. 1h later, before both your entries are realized on the tracker.

To see if your tracker is dual-stack, you can
```sh
dig A yourprivatetracker.me
dig AAAA yourprivatetracker.me
# replacing yourprivatetracker.me with your tracker's domain name 
```
and see if you get addresses for both record types.

**Comment by doomtop** [comment]
> one thing that would be of interest to this is to know whether you're on a dual-stack node (both IPv4 and IPv6).

@JustinTArthur Not dual-stack, IPv4 only.

**Comment by JustinTArthur** [comment]
That's an interesting anecdote re the network stack getting poisoned, @ilikepeaches. It rings similar to #4205 that popped up earlier in 4.x development.

**Comment by likeaross** [comment]
adding the following in my settings.json seems to fix the problem for me 
`"bind-address-ipv6": "",`
I had it set to "::" prior.

**Comment by ckerr** [comment]
I'm looking for testers on a download performance PR @ #5176 and thought of the people in this ticket.

The goal of this PR is to improve how many peers in a swarm Transmission will download from simultaneously and that's the kind feedback I'd like to hear about in the comments section of 5176.

If that sounds interesting interesting and you'd like to test, @Coeur has written up [a nice howto](https://github.com/transmission/transmission/blob/main/docs/Testing-Transmission.md) on testing CI builds.

**Comment by Coeur** [comment]
@EDavidzar and @ilikepeaches, since you both mentioned download speeds of 60 MB/s, would you know how I can find torrents with such swarm performances? Then I may be able to also help testing #5176.

**Comment by garoto** [comment]
It shouldn't be too hard to find public torrents that will download in excess of 50MB/s in 2023. Just tried a FreeBSD and a Debian one and they averaged 43.1 MB/s and 48.1 MB/s respectively. Too bad they aren't large enough I guess.

**Comment by EDavidzar** [comment]
Testing in linux  with Transmission 4.0.1 25MB/s is the highest speed i`m getting.

**Comment by GaryElshaw** [comment]
Have you tried the proposed changes? #5176

**Comment by EDavidzar** [comment]
I have not been able to, but I hope to test it between tomorrow and this weekend.

**Comment by EDavidzar** [comment]
tested another time transmission 4.0.1 Ubuntu linux 24MB/s highest speed and 16MB/s to 17MB/s 4 min download time and  tested nightly build of transmission with "bind-address-ipv6": "::", 117,1 MB/s highest speed download 1 minute total time and download and 82 MB/s  to 111 MB/s aprox.

**Comment by tearfur** [comment]
@JustinTArthur Does it really?

> In v4, you will announce once from an IPv4 source address and once from your IPv6 source address.

I have both IPv4 and IPv6 connectivity. However, on 4.0.1, I can only see one announcement via IPv6 when I tried to start downloading the [Debian 11.6.0 amd64 DVD ISO torrent](https://cdimage.debian.org/debian-cd/current/amd64/bt-dvd/debian-11.6.0-amd64-DVD-1.iso.torrent).

# Observations

## tcpdump

Command output when running `tcpdump -nn -A dst host bttracker.debian.org` just before starting the download and terminating it after Transmission got peers:

```
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on eno1, link-type EN10MB (Ethernet), snapshot length 262144 bytes
12:29:59.285341 IP6 REDACTED.33876 > 2001:6b0:e:2018::158.6969: Flags [S], seq 2470523983, win 2880, options [mss 1440,sackOK,TS val 631331172 ecr 0,nop,wscale 0], length 0
`....(.@*....ANJ........ ..... ........X.T.9.A4O.......@...........
%.Yd........
12:29:59.321447 IP6 REDACTED.33876 > 2001:6b0:e:2018::158.6969: Flags [.], ack 2714962884, win 2880, options [nop,nop,TS val 631331208 ecr 3879232492], length 0
`.... .@*....ANJ........ ..... ........X.T.9.A4P.......@.......
%.Y..8c.
12:29:59.321652 IP6 REDACTED.33876 > 2001:6b0:e:2018::158.6969: Flags [P.], seq 0:343, ack 1, win 2880, options [nop,nop,TS val 631331208 ecr 3879232492], length 343
`....w.@*....ANJ........ ..... ........X.T.9.A4P.......@.2.....
%.Y..8c.GET /announce?info_hash=%07c%B7W4%19V%B9%BFY%94%E4%A9%92%C3do%9F%19%BD&peer_id=-TR4010-kclow8z4010n&port=51413&uploaded=0&downloaded=0&left=3909091328&numwant=80&key=3865989059&compact=1&supportcrypto=1&event=started HTTP/1.1
Host: bttracker.debian.org:6969
User-Agent: Transmission/4.0.1
Accept: */*
Accept-Encoding: deflate, gzip, br


12:29:59.357774 IP6 REDACTED.33876 > 2001:6b0:e:2018::158.6969: Flags [.], ack 802, win 2403, options [nop,nop,TS val 631331244 ecr 3879232549], length 0
`.... .@*....ANJ........ ..... ........X.T.9.A5.......  c.......
%.Y..8d%
12:29:59.398703 IP6 REDACTED.33876 > 2001:6b0:e:2018::158.6969: Flags [.], ack 803, win 2403, options [nop,nop,TS val 631331285 ecr 3879232549], length 0
`.... .@*....ANJ........ ..... ........X.T.9.A5.......  c.......
%.Y..8d%
12:29:59.938964 IP6 REDACTED.33876 > 2001:6b0:e:2018::158.6969: Flags [F.], seq 343, ack 803, win 2403, options [nop,nop,TS val 631331825 ecr 3879232549], length 0
`.... .@*....ANJ........ ..... ........X.T.9.A5.......  c.......
%.[..8d%
^C
6 packets captured
26 packets received by filter
0 packets dropped by kernel
```

## Testing IPv4 and IPv6 connectivity to tracker

`ping -4 -c4 bttracker.debian.org`:

```
PING  (130.239.18.158) 56(84) bytes of data.
64 bytes from abdulaziz.acc.umu.se (130.239.18.158): icmp_seq=1 ttl=49 time=35.2 ms
64 bytes from abdulaziz.acc.umu.se (130.239.18.158): icmp_seq=2 ttl=49 time=35.3 ms
64 bytes from abdulaziz.acc.umu.se (130.239.18.158): icmp_seq=3 ttl=49 time=35.3 ms
64 bytes from abdulaziz.acc.umu.se (130.239.18.158): icmp_seq=4 ttl=49 time=35.3 ms

---  ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 3003ms
rtt min/avg/max/mdev = 35.214/35.265/35.326/0.039 ms
```

`ping -6 -c4 bttracker.debian.org`:

```
PING bttracker.debian.org(abdulaziz.acc.umu.se (2001:6b0:e:2018::158)) 56 data bytes
64 bytes from abdulaziz.acc.umu.se (2001:6b0:e:2018::158): icmp_seq=1 ttl=49 time=36.5 ms
64 bytes from abdulaziz.acc.umu.se (2001:6b0:e:2018::158): icmp_seq=2 ttl=49 time=36.4 ms
64 bytes from abdulaziz.acc.umu.se (2001:6b0:e:2018::158): icmp_seq=3 ttl=49 time=36.4 ms
64 bytes from abdulaziz.acc.umu.se (2001:6b0:e:2018::158): icmp_seq=4 ttl=49 time=36.4 ms

--- bttracker.debian.org ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 3003ms
rtt min/avg/max/mdev = 36.413/36.429/36.463/0.019 ms
```

`ping -c4 bttracker.debian.org`:

```
PING bttracker.debian.org(abdulaziz.acc.umu.se (2001:6b0:e:2018::158)) 56 data bytes
64 bytes from abdulaziz.acc.umu.se (2001:6b0:e:2018::158): icmp_seq=1 ttl=49 time=36.4 ms
64 bytes from abdulaziz.acc.umu.se (2001:6b0:e:2018::158): icmp_seq=2 ttl=49 time=36.5 ms
64 bytes from abdulaziz.acc.umu.se (2001:6b0:e:2018::158): icmp_seq=3 ttl=49 time=36.4 ms
64 bytes from abdulaziz.acc.umu.se (2001:6b0:e:2018::158): icmp_seq=4 ttl=49 time=36.7 ms

--- bttracker.debian.org ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 3003ms
rtt min/avg/max/mdev = 36.396/36.486/36.663/0.105 ms
```

**Comment by Coeur** [comment]
@ckerr looking at the discussion, could we have a configurable option so that users can decide by themselves when they want ipv4-only announcement, ipv6-only announcement, or dual ipv4-ipv6 announcement? That might help to pinpoint some perf issues for testers.

**Comment by tearfur** [comment]
Following my previous comment, I set `"bind-address-ipv6": ""` instead of the default `"bind-address-ipv6": "::"` as @likeaross suggested. It seems to have fixed the issue.

Now I can see a `GET /announce` request via both IPv4 and IPv6 in `tcpdump -nn -A dst host bttracker.debian.org`:
```
listening on eno1, link-type EN10MB (Ethernet), snapshot length 262144 bytes
15:23:19.871024 IP REDACTED.39264 > 130.239.18.158.6969: Flags [S], seq 2622765832, win 2920, options [mss 1460,sackOK,TS val 1347351646 ecr 0,nop,wscale 0], length 0
E..<..@.@.@.t........`.9.T;........h.;.........
PN.^........
15:23:19.871420 IP6 REDACTED.55714 > 2001:6b0:e:2018::158.6969: Flags [S], seq 2705395785, win 2880, options [mss 1440,sackOK,TS val 641731758 ecr 0,nop,wscale 0], length 0
`.5..(.@*....ANJ........ ..... ........X...9.A.I.......@...........
&@..........
15:23:19.906317 IP REDACTED.39264 > 130.239.18.158.6969: Flags [.], ack 3768636573, win 2920, options [nop,nop,TS val 1347351681 ecr 3844689501], length 0
E..4..@.@.@.t........`.9.T;     .......h.3.....
PN...)N]
15:23:19.906451 IP REDACTED.39264 > 130.239.18.158.6969: Flags [P.], seq 0:343, ack 1, win 2920, options [nop,nop,TS val 1347351681 ecr 3844689501], length 343
E.....@.@.?.t........`.9.T;     .......h.......
PN...)N]GET /announce?info_hash=%07c%B7W4%19V%B9%BFY%94%E4%A9%92%C3do%9F%19%BD&peer_id=-TR4010-1tzehyvdairn&port=51413&uploaded=0&downloaded=0&left=3909091328&numwant=80&key=3899210886&compact=1&supportcrypto=1&event=started HTTP/1.1
Host: bttracker.debian.org:6969
User-Agent: Transmission/4.0.1
Accept: */*
Accept-Encoding: deflate, gzip, br


15:23:19.906575 IP6 REDACTED.55714 > 2001:6b0:e:2018::158.6969: Flags [.], ack 3470517762, win 2880, options [nop,nop,TS val 641731793 ecr 3889633068], length 0
`.5.. .@*....ANJ........ ..... ........X...9.A.J.......@.......
&@.....,
15:23:19.906681 IP6 REDACTED.55714 > 2001:6b0:e:2018::158.6969: Flags [P.], seq 0:343, ack 1, win 2880, options [nop,nop,TS val 641731793 ecr 3889633068], length 343
`.5..w.@*....ANJ........ ..... ........X...9.A.J.......@.2.....
&@.....,GET /announce?info_hash=%07c%B7W4%19V%B9%BFY%94%E4%A9%92%C3do%9F%19%BD&peer_id=-TR4010-1tzehyvdairn&port=51413&uploaded=0&downloaded=0&left=3909091328&numwant=80&key=3899210886&compact=1&supportcrypto=1&event=started HTTP/1.1
Host: bttracker.debian.org:6969
User-Agent: Transmission/4.0.1
Accept: */*
Accept-Encoding: deflate, gzip, br


15:23:19.941825 IP6 REDACTED.55714 > 2001:6b0:e:2018::158.6969: Flags [.], ack 838, win 2511, options [nop,nop,TS val 641731828 ecr 3889633134], length 0
`.5.. .@*....ANJ........ ..... ........X...9.A.....G..  ........
&@.....n
15:23:19.941839 IP REDACTED.39264 > 130.239.18.158.6969: Flags [.], ack 620, win 2476, options [nop,nop,TS val 1347351716 ecr 3844689567], length 0
E..4..@.@.@.t........`.9.T<`......      ..3.....
PN...)N.
15:23:19.941964 IP REDACTED.39264 > 130.239.18.158.6969: Flags [F.], seq 343, ack 621, win 2476, options [nop,nop,TS val 1347351716 ecr 3844689567], length 0
E..4..@.@.@.t........`.9.T<`... ..      ..3.....
PN...)N.
15:23:19.942024 IP6 REDACTED.55714 > 2001:6b0:e:2018::158.6969: Flags [F.], seq 343, ack 839, win 2511, options [nop,nop,TS val 641731829 ecr 3889633134], length 0
`.5.. .@*....ANJ........ ..... ........X...9.A.....H..  ........
&@.....n
^C
10 packets captured
12 packets received by filter
0 packets dropped by kernel
```

**Comment by Coeur** [comment]
The double colon is defined here:
https://github.com/transmission/transmission/blob/dabd29f2b289e55ad2740ec4e4b22385fef72018/libtransmission/session-settings.h#L23

If one believes it's a bug, feel free to make a pull request to fix it. You don't even need to know Git for that: just click the edit button and submit a PR. But please first test that it works.
![Capture d’écran 2023-03-12 à 15 43 41](https://user-images.githubusercontent.com/839992/224531365-fbdb174c-614e-4354-89cd-a174c79c5f37.png)

**Comment by ckerr** [comment]
I am going to close this ticket because I'm eagerly awaiting "Transfer performance remains significantly degraded in 4.0.2 compared to 3.0.0" :tada:
